tuf/api/metadata.py

```diff
@@ -539,6 +539,25 @@ def to_securesystemslib_key(self) -> Dict[str, Any]:
             "keyval": self.keyval,
         }
 
+    @classmethod
+    def from_securesystemslib_key(cls, key_dict: Dict[str, Any]) -> "Key":
+        """
+        Creates a Key object from a securesystemlib key dict representation
+        removing the private key from keyval.
+        """
+        key_meta = sslib_keys.format_keyval_to_metadata(
+            key_dict["keytype"],
+            key_dict["scheme"],
+            key_dict["keyval"],
+        )
+        return cls(
+            key_dict["keyid"],
+            key_meta["keytype"],
+            key_meta["scheme"],
+            key_meta["keyval"],
+            key_dict,
+        )
```

Comment:
this is interesting -- I was half expecting we'd have to make assumptions about keyval contents in this function but you found an sslib function that mostly does what you need... I think this looks reasonable except on the last line you add the whole key_dict (including the secret) as "unrecognised fields", oops. No need for unrecognised_fields here.

You _could_ modify the test so it looks for the secret in the serialized format of the key to be sure... but that sounds a bit much. Checking to_dict() output manually (to see that the above issue is fixed) should be fine.